app/models/v2/policy.rb

```diff
@@ -10,6 +10,8 @@ class Policy < ApplicationRecord
     belongs_to :profile, class_name: 'V2::Profile'
     has_one :security_guide, through: :profile, class_name: 'V2::SecurityGuide'
     has_many :tailorings, class_name: 'V2::Tailoring', dependent: :destroy
+    has_many :tailoring_rules, through: :tailorings, class_name: 'V2::TailoringRule', dependent: :destroy
+    has_many :rules, through: :tailoring_rules, class_name: 'V2::Rule', dependent: :destroy
```

Comment:
You should not delete any rules when deleting a link:
```suggestion
    has_many :rules, through: :tailoring_rules, class_name: 'V2::Rule'
```